lib/membrane/video_compositor/scene.ex

```diff
@@ -0,0 +1,25 @@
+defmodule Membrane.VideoCompositor.Scene do
+  @moduledoc """
+  Structure representing a top level specification of what is Video Compositor
+  supposed to render.
+
+  The main part of the Scene are `Membrane.VideoCompositor.Scene.Object`s
+  and interactions between them. There are three kinds of Objects:
+  - `Membrane.VideoCompositor.Scene.Object.Texture` - single input object,
+  taking frames and applying a series of transformations onto it.
+  - `Membrane.VideoCompositor.Scene.Object.Layout` - combining
+  frames from multiple inputs into a single output.
+  - `Membrane.VideoCompositor.Scene.Object.InputVideo` - which maps
+  an input pad of element into Scene object.
+  """
+
+  alias Membrane.VideoCompositor.Scene.Object
+
+  @enforce_keys [:objects, :output]
+  defstruct @enforce_keys
+
+  @type t :: %__MODULE__{
+          objects: [{Object.name(), Object.t()}],
+          output: Object.name()
+        }
```

Comment:
Maybe we can leave a note here, that if all `object names` are atoms, `objects` can simply be a keyword list

Reply:
@Janix4000 What do you think?

Reply:
I don't understand what you're suggesting @WojciechBarczynski

Reply:
My original suggestion was just to leave in `@typedoc` info, that if all names of objects are simply atoms, then we can define `objects: some_keyword_list` as an field in the `scene` struct (where `Keyword.keyword?(some_keyword_list)` is `true`). Then Dominik suggested removing `tuple` from the `Object.name` type definition, such that we can always use a keyword list here (at least from what have I understood)

Reply:
There is no significant advantage to restricting users to use only atoms as object ids. From our perspective, there is no difference in the types of ids. From the users' perspective, it is very convenient to use tuples of atom names and integer indices instead of generated atoms. Additionally, it is not recommended to generate atoms in runtime.

Reply:
it actually makes no difference in the code whatsoever, we will hold names as `rustler::Term`s on the rust side, which means anything can be a name

Reply:
Ok